src/tests/test_des.cpp

```diff
@@ -0,0 +1,37 @@
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/nar/des/des.h"
+#include "all_csv_configs.h"
+#include "config/names.h"
+
+namespace tests {
+
+class DESTest : public ::testing::Test {
+protected:
+    static algos::StdParamsMap GetParamMap(CSVConfig const& csv_config, double minsup,
+                                           double minconf, int popSize, int evalNum,
+                                           double crossProb, double diffScale,
+                                           algos::des::DifferentialStrategy diffStrategy) {
+        using namespace config::names;
+        return {{kCsvConfig, csv_config},          {kMinimumSupport, minsup},
+                {kMinimumConfidence, minconf},     {kPopulationSize, popSize},
+                {kMaxFitnessEvaluations, evalNum}, {kCrossoverProbability, crossProb},
+                {kDifferentialScale, diffScale},   {kDifferentialStrategy, diffStrategy}};
+    }
+
+    template <typename... Args>
+    static std::unique_ptr<algos::des::DES> CreateAlgorithmInstance(Args&&... args) {
+        return algos::CreateAndLoadAlgorithm<algos::des::DES>(
+                GetParamMap(std::forward<Args>(args)...));
+    }
+};
+
+TEST_F(DESTest, LaunchTest) {
+    auto algorithm = CreateAlgorithmInstance(kAbalone, 0.5, 0.6, 100, 100, 0.9, 0.5,
+                                             algos::des::DifferentialStrategy::rand1Bin);
+    algorithm->Execute();
+    SUCCEED();
+}
+
```

Comment:
Add tests, checking that algorithm works fine:
- test that NARs that should hold are found
- test that their qualities are calculated correctly (fitness, support, confidence)

Reply:
This algorithm converges on non-useful NARs as it is right now. Just added a test that checks if it mined an exact expected set of NARs. Might help to find out when a change unexpectedly affected algo behaviour.